packages/klighd-core/src/options/general-panel.tsx

```diff
@@ -0,0 +1,147 @@
+    }
+
+    render(): VNode {
+        return (
+            <div>
+                <div classNames="options__section">
+                    <h5 classNames="options__heading">Quick Actions</h5>
+                    <div classNames="options__button-group">
+                        {this.quickActions.map((action) => (
+                            <button
+                                title={action[1]}
+                                classNames="options__icon-button"
+                                on-click={() => this.handleQuickActionClick(action[0])}
+                            >
+                                {action[2]}
+                            </button>
+                        ))}
+                    </div>
+                </div>
+                <div classNames="options__section">
+                    <h5 classNames="options__heading">Synthesis</h5>
+                    <SynthesisPicker
+                        currentId={this.synthesesRegistry.currentSynthesisID}
+                        syntheses={this.synthesesRegistry.syntheses}
+                        onChange={this.handleSynthesisPickerChange.bind(this)}
+                    />
+                </div>
+                <div classNames="options__section">
+                    <h5 classNames="options__heading">Render Options</h5>
+                    {this.optionsRenderer.renderRenderOptions(
+                        this.renderOptionsRegistry.allRenderOptions
+                    )}
+                </div>
+                <div classNames="options__section">
+                    <h5 classNames="options__heading">Preferences</h5>
+                    <CheckOption
+                        id="resizeToFit"
+                        name="Resize To Fit"
+                        value={this.preferencesRegistry.preferences.resizeToFit}
+                        onChange={this.handlePreferenceChange.bind(this, "resizeToFit")}
+                    />
+                    <CheckOption
+                        id="forceLightBackground"
+                        name="Use Light Background"
+                        value={this.preferencesRegistry.preferences.forceLightBackground}
+                        onChange={this.handlePreferenceChange.bind(this, "forceLightBackground")}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private handleSynthesisPickerChange(newId: string) {
+        this.actionDispatcher.dispatch(new SetSynthesisAction(newId));
+    }
+
+    private handlePreferenceChange(key: string, newValue: any) {
+        this.actionDispatcher.dispatch(new SetPreferencesAction({ [key]: newValue }));
+    }
+
+    private handleQuickActionClick(type: PossibleAction) {
+        const action = this.quickActions.find((a) => a[0] === type)?.[3];
+
+        if (!action) return;
+
+        this.actionDispatcher.dispatch(action);
+    }
+
+    get icon(): string {
+        // Icon-source: https://feathericons.com/?query=settings
```

Comment:
I switched the icons to use the package. To render them in the sidebar, the replace method has be used, since Snabbdom can not convert the SVG strings to JSX.

Using the icon package enabled a refactor of the sidebar, which minimizes the rerender logic significantly. The refactor is applied in https://github.com/kieler/klighd-vscode/commit/daa33e79abd9b21f250ef00e4a74302850f8a606